src/OrbitGl/Track.cpp

```diff
@@ -28,70 +29,7 @@ Track::Track(CaptureViewElement* parent, const orbit_gl::TimelineInfoInterface*
       timeline_info_(timeline_info),
       module_manager_(module_manager),
       capture_data_(capture_data) {
-  collapse_toggle_ = std::make_shared<TriangleToggle>(
-      [this](bool is_collapsed) { OnCollapseToggle(is_collapsed); }, viewport, layout, this);
-}
-
-std::vector<Vec2> GetRoundedCornerMask(float radius, uint32_t num_sides) {
-  std::vector<Vec2> points;
-  points.emplace_back(0.f, 0.f);
-  points.emplace_back(0.f, radius);
-
-  float increment_radians = 0.5f * kPiFloat / static_cast<float>(num_sides);
-  for (uint32_t i = 1; i < num_sides; ++i) {
-    float angle = kPiFloat + static_cast<float>(i) * increment_radians;
-    points.emplace_back(radius * cosf(angle) + radius, radius * sinf(angle) + radius);
-  }
-
-  points.emplace_back(radius, 0.f);
-  return points;
-}
-
-std::vector<Vec2> RotatePoints(const std::vector<Vec2>& points, float rotation) {
-  float cos_r = cosf(kPiFloat * rotation / 180.f);
-  float sin_r = sinf(kPiFloat * rotation / 180.f);
-  std::vector<Vec2> result;
-  for (const Vec2& point : points) {
-    float x_rotated = cos_r * point[0] + sin_r * point[1];
-    float y_rotated = sin_r * point[0] - cos_r * point[1];
-    result.emplace_back(x_rotated, y_rotated);
-  }
-  return result;
-}
-
-void Track::DrawTriangleFan(Batcher& batcher, const std::vector<Vec2>& points, const Vec2& pos,
-                            const Color& color, float rotation, float z) {
-  if (points.size() < 3) {
-    return;
-  }
-
-  std::vector<Vec2> rotated_points = RotatePoints(points, rotation);
-  Vec3 position(pos[0], pos[1], z);
-  Vec3 pivot = position + Vec3(rotated_points[0][0], rotated_points[0][1], z);
-
-  Vec3 vertices[2];
-  vertices[0] = position + Vec3(rotated_points[1][0], rotated_points[1][1], z);
-
-  for (size_t i = 1; i < rotated_points.size() - 1; ++i) {
-    vertices[i % 2] = position + Vec3(rotated_points[i + 1][0], rotated_points[i + 1][1], z);
-    Triangle triangle(pivot, vertices[i % 2], vertices[(i + 1) % 2]);
-    batcher.AddTriangle(triangle, color, shared_from_this());
-  }
-}
-
-void Track::UpdatePositionOfCollapseToggle() {
-  const float label_height = layout_->GetTrackTabHeight();
-  const float half_label_height = 0.5f * label_height;
-  const float x0 = GetPos()[0] + layout_->GetTrackTabOffset() +
-                   layout_->GetTrackIndentOffset() * indentation_level_;
-  const float button_offset = layout_->GetCollapseButtonOffset();
-  const float toggle_y_pos = GetPos()[1] + half_label_height;
-  Vec2 toggle_pos = Vec2(x0 + button_offset, toggle_y_pos);
-
-  float size = layout_->GetCollapseButtonSize(indentation_level_);
-  collapse_toggle_->SetWidth(size);
-  collapse_toggle_->SetHeight(size);
-  collapse_toggle_->SetPos(toggle_pos[0], toggle_pos[1]);
+  header_ = std::make_shared<orbit_gl::TrackHeader>(this, viewport, layout);
```

Comment:
So why is this a shared_ptr?

Reply:
Because the `PickingManager` expects shared pointers to picked elements. We may want to get rid of this in the future, but we've had this discussion in `PickingManager` already and AFAIK concluded that we want to keep it like this for now.

Reply:
I see....